core/src/main/scala/org/ensime/core/javac/Helpers.scala

```diff
@@ -47,32 +47,31 @@ trait Helpers extends UnsafeHelpers with SLF4JLogging {
     Some(JavaFqn(front.mkString("."), back.mkString("."), None))
   }
 
-  def fqn(info: CompilationInfo, el: Element): Option[JavaFqn] = {
-    val kind = el.getKind
-    if (kind == ElementKind.LOCAL_VARIABLE || kind == ElementKind.PARAMETER) {
-      Some(JavaFqn(None, None, Some(el.getSimpleName.toString)))
-    } else if (kind == ElementKind.CONSTRUCTOR || kind == ElementKind.ENUM_CONSTANT ||
-      kind == ElementKind.METHOD || kind == ElementKind.FIELD) {
-      Option(el.getEnclosingElement).flatMap(fqn(info, _)).map(_.copy(fieldOrMethod = Some(el.toString)))
-    } else {
-      parseFqnAsClass(el.toString)
+  private val isSimpleKind = Set(ElementKind.LOCAL_VARIABLE, ElementKind.PARAMETER)
+  private val isComplexKind = Set(ElementKind.CONSTRUCTOR, ElementKind.ENUM_CONSTANT, ElementKind.METHOD, ElementKind.FIELD)
+
+  def fqn(c: Compilation, el: Element): Option[JavaFqn] = {
+    el.getKind match {
+      case k if isSimpleKind(k) => Some(JavaFqn(None, None, Some(el.getSimpleName.toString)))
```

Comment:
I have a feeling that the difference might not be measurable, but I might be wrong...